AAInstrumentation/maximal_relaxation.py

```diff
@@ -0,0 +1,311 @@
+from core import AAInstrumentationDriver, register_arguments
+from multiprocessing import Pool
+import os
+from pathlib import Path
+import subprocess
+import hashlib
+import argparse
+import sys
+from dataclasses import dataclass
+
+
+@dataclass
+class MaximalRelaxationDriver(AAInstrumentationDriver):
+    original_hash: str = 0
+    instrument_aa_recursively: bool = False
+
+    def get_groundtruth_hash(self, file_name: Path):
+        """Returns the hash of the groundtruth file."""
+
+        self.run_step_single_func(
+            file_name,
+            0,
+            [],
+            self.instrument_aa_recursively,
+        )
+        self.assemble_file(
+            self.instr_dir.joinpath(
+                file_name.parent, str(0) + str(file_name.stem) + ".bc"
+            ),
+            self.instr_dir.joinpath(
+                file_name.parent, str(0) + str(file_name.stem) + ".o"
+            ),
+        )
+
+        return self.get_hash(
+            self.instr_dir.joinpath(
+                file_name.parent, str(0) + str(file_name.stem) + ".o"
```

Comment:
(almost) the same path is repeated 3 times, factor it out and use `.with_suffix` for the `.bc`?